panther_lights/test/test_led_segment.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdint>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <gtest/gtest.h>
+#include <yaml-cpp/yaml.h>
+
+#include <panther_lights/led_segment.hpp>
+
+class TestLEDSegment : public testing::Test
+{
+public:
+  TestLEDSegment();
+  ~TestLEDSegment() {}
+
+protected:
+  std::shared_ptr<panther_lights::LEDSegment> led_segment_;
+
+  const float controller_freq = 50.0;
+  const std::size_t segment_led_num_ = 10;
+};
+
+TestLEDSegment::TestLEDSegment()
+{
+  auto segment_desc =
+    YAML::Load("{led_range: 0-" + std::to_string(segment_led_num_ - 1) + ", channel: 1}");
+  led_segment_ = std::make_shared<panther_lights::LEDSegment>(segment_desc, controller_freq);
+}
+
+// this method may be moved to utils
```

Comment:
It's a good idea. Should we do it now?

Reply:
Looked at it and it may be a bit tricky - test_utils have dependencie on rclcpp. There is no point in adding rclcpp dependencies in here. The best solution, I guess, is to devide test_utils into 2 files(for ros and non ros utils). This however makes need for changes in other repositories using current test_utils and I dont nkow if it is a good idea to make them in this PR

Reply:
ok, so it's up to you